### PR TITLE
Fix #41 unable to integrate test target

### LINKED ISSUE #41: Unable to integrate into iOS unit test target
I've been struggling for a while to integrate the testing library into a unit test target, but I haven't found a solution that works yet.

Given a pod file like the following : 

```
platform :ios, '10.0'

use_frameworks!

target 'tacx' do
    
    pod 'RZBluetooth/Mock'
    
    target 'tacxTests' do
        inherit! :search_paths
        pod 'RZBluetooth/Test'
    end

end
```

A unit testing file: 
```objective-c
@import RZBluetooth;

@interface tacxTests : RZBSimulatedTestCase
@end

@implementation tacxTests
@end
```

Everything compiles just fine, but when running the test, it prints a message stating :

> 2016-11-20 19:05:59.112 tacx[95628:5961223] Failed to load test bundle from file:///Users/xxxxxxx/Library/Developer/Xcode/DerivedData/tacx-ghqgcpxbcphxnmhievkvngfndhzq/Build/Intermediates/CodeCoverage/Products/Debug-iphonesimulator/tacx.app/PlugIns/tacxTests.xctest/: Error Domain=NSCocoaErrorDomain Code=3588 "dlopen(/Users/xxxxxxx/Library/Developer/Xcode/DerivedData/tacx-ghqgcpxbcphxnmhievkvngfndhzq/Build/Intermediates/CodeCoverage/Products/Debug-iphonesimulator/tacx.app/PlugIns/tacxTests.xctest/tacxTests, 265): Symbol not found: _OBJC_CLASS_$_RZBSimulatedTestCase
  Referenced from: /Users/xxxxxxx/Library/Developer/Xcode/DerivedData/tacx-ghqgcpxbcphxnmhievkvngfndhzq/Build/Intermediates/CodeCoverage/Products/Debug-iphonesimulator/tacx.app/PlugIns/tacxTests.xctest/tacxTests
  Expected in: /Users/xxxxxxx/Library/Developer/CoreSimulator/Devices/D4A4BA5A-3023-4B18-B912-EE1FCDA1835D/data/Containers/Bundle/Application/2060D194-B119-4FEA-A8B1-917686855AB4/tacx.app/Frameworks/RZBluetooth.framework/RZBluetooth
 in

As you can see by the `Referenced from` and `Expected in` lines, Xcode sees a symbol from the  `RZBluetooth` framework (`_OBJC_CLASS_$_RZBSimulatedTestCase`) and expects it to be located in `/tacx.app/Frameworks/RZBluetooth.framework/RZBluetooth`, which is the app's version of `RZBluetooth` (created by `pod 'RZBluetooth/Mock'` in the podfile). This causes the app to crash because it can not find a the symbols for `RZBSimulatedTestCase`.

Are you aware of any way to explicitly importing the unit test's version of the library? (that is the version of the library that contains the `RZBSimulatedTestCase`class?

### PR DESCRIPTION
**Pull request by joshuatbrown**
This fixes #41. However, due to the way we fixed it, the Tests target in RZBluetooth no longer compiles. We've tried various ways to fix the problem but haven't found a solution yet. I'm hoping that by opening this PR, we'll get some help with the Tests target.

Here's what we did:

- created a new podspec for RZBluetoothTest that contains the files that used to be part of the `RZBluetooth/Test` subspec
- changed the imports in `RZBSimulatedTestCase` since they now depend on the main `RZBluetooth/Mock` spec (and therefore need to be imported with `RZBluetooth/...`)

The second bullet is where the problem lies. By importing with `RZBluetooth/...`, we've broken the test target which does not have those files in a separate framework. But importing this way is required so that the new `RZBluetoothTest` podspec builds in CocoaPods.

### CODE REVIEW

File: RZBluetoothTests/RZBSimulatedTestCase.m [hunk 1]
```diff
@@ -7,11 +7,11 @@
 //
 
 #import "RZBTestDefines.h"
-#import "RZMockBluetooth.h"
+#import "RZBluetooth/RZMockBluetooth.h"
 #import "RZBSimulatedTestCase.h"
 #import "NSRunLoop+RZBWaitFor.h"
-#import "RZBCentralManager+Private.h"
-#import "RZBLog.h"
+#import "RZBluetooth/RZBCentralManager+Private.h"
```
Comment (KingOfBrian) [inline RZBluetoothTests/RZBSimulatedTestCase.m]: It's been a while... Why do we need the private header? Ideally we wouldn't, but if we can safely remove it we should.

File: RZBluetoothTest.podspec [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+Pod::Spec.new do |s|
+  s.name         = "RZBluetoothTest"
+  s.version      = "1.2.2"
+  s.summary      = "A Core Bluetooth helper library to simplify the development and testing of Core Bluetooth applications."
```
Comment (KingOfBrian) [inline RZBluetoothTest.podspec]: The summary should state that this pod is specifically so simplify testing bluetooth with XCTest and RZBluetooth mocks.

**Review by KingOfBrian — CHANGES_REQUESTED** [review]
Looks good @joshuatbrown. We do need to get the tests working though. I may get some time to look into it this week.

Also, it'd be great to do another pod for the Mocks if you're interested!

File: RZBluetoothTests/RZBSimulatedTestCase.m (hunk 1 above)
Comment (mplorentz) [reply RZBluetoothTests/RZBSimulatedTestCase.m]: Looks like it exposes the `RZBCommandDispatch *dispatch` property which is used in `waitForQueueFlush` and required for `RZBAssertCommandCount()`.

**Comment by mplorentz** [comment]
@KingOfBrian I think I made all the changes you requested. I also updated the RZBluetoothExample app to use the RZBluetoothTest podspec. Can you re-review?

**Comment by joshbrown-eg** [comment]
This looks good — would love to get this merged.

**Comment by cpatterson-lilly** [comment]
Does RZBluetoothMock need similar treatment to be broken out into its own podspec?

**Pull request closed without merging**